src/pudl/analysis/allocate_net_gen.py

```diff
@@ -196,43 +197,166 @@
 """
 
 
-# Two top-level functions (allocate & aggregate)
+def allocate_net_gen_asset_factory(
+    freq: Literal["AS", "MS"],
+    io_manager_key: str | None = None,
+) -> list[AssetsDefinition]:
+    """Build net gen allocation assets at yearly and monthly frequencies."""
```

Comment:
Maybe include fuel consumption as well as net generation in the description here.  I think we might want to change all of the references to `allocate_net_gen` to `allocate_gen_fuel` or something similar, since I think we're always allocating both fuel and generation now.